tests/test_cargo_test.rs

```diff
@@ -46,6 +46,44 @@ test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured
         RUNNING)));
 });
 
+test!(cargo_test_release {
+    let p = project("foo")
+        .file("Cargo.toml", basic_bin_manifest("foo").as_slice())
+        .file("src/foo.rs", r#"
+            fn hello() -> &'static str {
+                "hello"
+            }
+
+            pub fn main() {
+                println!("{}", hello())
+            }
+
+            #[test]
+            fn test_hello() {
+                assert_eq!(hello(), "hello")
+            }"#);
+
+    assert_that(p.cargo_process("build").argv("--release"), execs());
+    assert_that(&p.bin("foo"), existing_file());
+
+    assert_that(process(&p.bin("foo")).unwrap(),
+                execs().with_stdout("hello\n"));
+
+    assert_that(p.cargo("test").argv("--release"),
```

Comment:
Could you add the `-v` flag here to ensure that targets are compiled with the right flags? It's also fine to omit the above assertions around `cargo build` and wehther the binary `foo` exists.